src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+    private int randomNumber = random.nextInt(5);
+
+    public Figure getRandomFigure() {
+        switch (randomNumber) {
+            case 0:
+                return new Square(colorSupplier.getRandomColor(), random.nextDouble() * 10);
+            case 1:
+                return new Circle(colorSupplier.getRandomColor(), random.nextDouble() * 10);
+            case 2:
+                return new Rectangle(colorSupplier.getRandomColor(), random.nextDouble() * 10,
+                        random.nextDouble() * 10, random.nextDouble() * 10);
+            case 3:
+                return new RightTriangle(colorSupplier.getRandomColor(), random.nextDouble() * 10,
+                        random.nextDouble() * 10, random.nextDouble() * 10);
+            default:
+                return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                       random.nextDouble() * 10,
+                        random.nextDouble() * 10,
+                       random.nextDouble() * 10,
+                       random.nextDouble() * 10);
```

Comment:
Avoid passing expressions directly in the constructor for better readability. Create separate variables for the random values and pass them to the constructors.